pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -38,9 +43,18 @@ import (
 	"istio.io/istio/pkg/workloadapi"
 )
 
-// AmbientIndex maintains an index of ambient WorkloadInfo objects by various keys.
+type AmbientIndex interface {
```

Comment:
Or is my IDE just wrong and Controller implements it as well?

Reply:
This interface is the point of this PR :). The controller has a reference to it rather than a `AmbientIndexImpl`. This prevents the controller from accessing any of its internals directly.

Reply:
I see. we could move it to another package possible as well, then can use standard visibility

Reply:
Yeah I thought about that as well. Maybe we could address in a follow-on PR?